### ISSUE TITLE
Saving formats 

### ISSUE THREAD

**Issue by pedrobrs** [keras-team/keras#18523]
I used to save models with ModelCheckpoint callback and filename .h5.
Recently update tensorflow to version 2.13.

Now If save the model in h5 format, a warning is logged:
 UserWarning: You are saving your model as an HDF5 file via `model.save()`. This file format is considered legacy. We recommend using instead the native Keras format, e.g. `model.save('my_model.keras')`.

But If use keras, I hit this issue:
https://stackoverflow.com/questions/76701617/the-following-arguments-are-not-supported-with-the-native-keras-format-opti

and, if I use tf format , I get no logs for fit!
When checkpoint filename ends with .h5 I get a log like:
 absl - 1106/1106 - 41s - loss: 0.0014 - accuracy: 0.9996 - precision: 0.9998 - recall: 0.9996 - auc: 1.0000 - val_loss: 0.0046 - val_accuracy: 0.9978 - val_precision: 0.9987 - val_recall: 0.9984 - val_auc: 1.0000 - lr: 0.0010 - 41s/epoch - 37ms/step
 
 But with a filename .tf,  that log is not generated. 
 
 
 I would like to use a non legacy format but without exceptions nor logs missing.
What can I do?

**Comment by SuryanarayanaY** [comment]
@pedrobrs ,

We have observed similar behaviour reported in TF repo as well recently and this issue persists from 2.13 version onwards. While saving model with .keras format and if model have a callback say for eg. 

```
tf.keras.callbacks.ModelCheckpoint(
        "models/tmp_saved_model4" + "/{epoch}_model.keras", 
        save_best_only=True #making it to False or None works
)
```
It fails with similar error as reported in the attached SO link.

Similar issue was discussed in TF #[61999](https://github.com/tensorflow/tensorflow/issues/61999). Please have a look into it.

As per my understanding there is issue with `model.save('somename.keras', options=something)` that causing the failure.

Could you please check and confirm for your case what are the options that are getting passed to model.save() ?

**Comment by SuryanarayanaY** [comment]
Hi @pedrobrs ,

I believe this issue should have fixed by this PR #18545. The relevant tensorflow issue #https://github.com/tensorflow/tensorflow/issues/61999 also resolved with this PR.

Could you please check the issue with nightly and let us know if yet to resolve. Thanks

**Comment by rajveer43** [comment]
can I work on this?

**Comment by rajveer43** [comment]
can I work on this?

**Comment by rickiepark** [comment]
Hi @SuryanarayanaY 
Same issue is in TF 2.14, please check this [gist].(https://colab.research.google.com/drive/17Fbhpb5XJqRZURZm8wR9xyEM0wOf1Fdc)

**Comment by SuryanarayanaY** [comment]
Hi @rickiepark ,

Could you please check with nightly version as changes are not merged into nightly only?  Also your colab gist not accessible.

**Comment by steveepreston** [comment]
this issue exists in TensorFlow 2.16.1

**Comment by ghsanti** [comment]
@steevepreston

> this issue exists in TensorFlow 2.16.1


Can you share a minimal colab that reproduces the error?

**Comment by YMasterous** [comment]
This issue also exists in TensorFlow 2.17.0
